pkg/config/config_template.yaml

```diff
@@ -620,18 +620,18 @@ api_key:
 
 {{- if .Profiling -}}
 ## @param profiling - custom object - optional
-## Enter specific configurations for profiling.
+## Enter specific configurations for internal profiling.
```

Comment:
Maybe add an explicit note that this feature is used by Datadog to profile the agent itself during troubleshooting?

Reply:
agreed, this is worth a long explicit note (and maybe removing the link to the profiling doc URL https://docs.datadoghq.com/tracing/profiling/ as well). Can be addressed later in a docs-only PR.